systems/trajectory_optimization/hybrid_dircon.cc

```diff
@@ -108,14 +108,12 @@ HybridDircon<T>::HybridDircon(const MultibodyPlant<T>& plant,
       impulse_vars_.push_back(NewContinuousVariables(
           constraints_[i]->countConstraintsWithoutSkipping(),
           "impulse[" + std::to_string(i) + "]"));
-    }
-    //  Post-impact variables. Note: impulse variables declared below.
-    if (i > 0) {
```

Comment:
I'm not seeing the off-by-one bug here. Isn't some sort of `if` statement necessary here, since there is one fewer impact than the number of modes.?

The old code seems like it's going to build `v_post_impact_vars_substitute_` with all of the `v_post_impact_vars_by_mode` elements, from 0,...,n-2. Won't the new code try to add post impact variables for the last mode (which doesn't exist)?